service/stub_module.go

```diff
@@ -33,14 +33,8 @@ type StubModule struct {
 var _ Module = &StubModule{}
 
 // NewStubModule generates a Module for use in testing
-func NewStubModule() *StubModule {
-	return &StubModule{}
-}
-
-// Initialize fakes an init call on the module
-func (s *StubModule) Initialize(host Host) error {
-	s.Host = host
-	return s.InitError
+func NewStubModule(host Host) *StubModule {
+	return &StubModule{Host: host}
```

Comment:
That is the conversation we had over chat yesterday. I expressed concerned around bootstrapping, where the service Builder is instantiating modules (with New) before the service owner has time to necessarily configure all the components. This is primarily a concern for user-defined modules (e.g. modules that don't live in this repository).

`AddModules` is called during `Build()` which is generally the first line of a user's `main` func. Between Build() and Start() they may hydrate more objects, connect to databases, etc.

@madhuravi wanted to remove it for now, and re-add it if we run into cases where we need a separate initialization phase.